docs/porting-to-ROS2.md

```diff
@@ -130,14 +170,181 @@ which is equivalent to
 
     const double vel_lim = declare_parameter<double>("vel_lim", 25.0);
 
+This allows to set the initial value e.g. via a parameter file.
+
+**NOTE** Calling `ros2 param set <NODE> vel_lim 1.234` after starting the node works but will not
+alter the member `vel_lim`! See the section below on *dynamic reconfigure* to achieve that.
+
+### dynamic reconfigure
+Dynamic reconfigure as it existed in ROS1 does not exist anymore in ROS2 and can be achieved by
+simpler means using a parameter callback.
+
+#### cfg files
+
+Remove the package's `.cfg` file and associated `cfg/` subdirectory.
+
+#### header file
+
+In the header file, remove includes of `dynamic_reconfigure` and the node-specific config file; e.g.
+
+```diff
+-#include <dynamic_reconfigure/server.h>
+-#include <mpc_follower/MPCFollowerConfig.h>
+```
+
+you need to set a parameter handler and callback function:
+
+```c++
+OnSetParametersCallbackHandle::SharedPtr set_param_res_;
+rcl_interfaces::msg::SetParametersResult paramCallback(const std::vector<rclcpp::Parameter> & parameters);
+```
+
+If there are many parameters (rule of thumb: more than 2), it is more practical to group them in a
+struct defined within the node's declaration:
+
+```c++
+class MPCFollower : public rclcpp::Node
+{
+
+  struct MPCParam
+  {
+    int prediction_horizon;
+    ...
+  } mpc_param;
+
+};
+
+```
+
+Add a method to declare all the parameters
+
+    void declareMPCparameters();
+
+#### implementation file
+
+Write the following into the definition of the class that inherits from `rclcpp::Node`.
+
+A few macro and a utility function help keep the following code and void of redundancy
```

Comment:
macros plural? Also, missing word after "code"

Reply:
Also, mention somewhere that "MPC" is just the name of the example?

Reply:
And perhaps mention that these are entirely optional/make sense when there is a lot of repetition. I think macros are discouraged by the Google Style guide at least.